app/scripts/views/components/pieChart/pieChart.js

```diff
@@ -121,6 +142,408 @@
+      }
+    };
+
+    content.initMainDivQtip = function(){
+      $('#' +v.opts.charDivId).qtip({
+        id: v.opts.charDivId+'-qtip',
+        style: {
+          classes: 'qtip-light qtip-rounded qtip-shadow forceZindex qtip-max-width study-view-qtip study-view-pie-label-qtip'
+        },
+        show: {event: "mouseover", solo: true, delay: 0, ready: true},
+        hide: {fixed:true, delay: 300, event: "mouseleave"},
+        //hide: false,
+        position: {my:'left center',at:'center right', viewport: $(window)},
+        content: '<div id="qtip-' + v.opts.charDivId + '-content-react">Loading....</div>',
+        events: {
+          render: function() {
+            updateCurrentLabels();
+            initReactData();
+            var data = $.extend(true, {}, reactTableData);
+            data.attributes[0].column_width = 140;
+            initReactTable('qtip-' + v.opts.charDivId + '-content-react',
+              data, {
+                tableWidth: 300,
+                pieLabelMouseEnterFunc: pieLabelMouseEnter,
+                pieLabelMouseLeaveFunc: pieLabelMouseLeave
+              });
+            pieLabelTableInitialized = true;
+          }
+        }
+      });
+    };
+
+
+    function initLabels() {
+      labelMetaData = initLabelInfo();
+      labels = $.extend(true, [] , labelMetaData);
+    }
+
+    function initLabelInfo() {
+      var _labelID = 0;
+      var _labels = [];
+      currentSampleSize = 0;
+
+      $('#' + v.opts.chartId + '>svg>g>g').each(function(){
+        var _labelDatum = {};
+        var _labelText = $(this).find('title').text();
+        var _color = $(this).find('path').attr('fill');
+        var _pointsInfo = $(this).find('path').attr('d').split(/[\s,MLHVCSQTAZ]/);
+        var _labelName = _labelText.substring(0, _labelText.lastIndexOf(":"));
+        var _labelValue = Number(_labelText.substring(_labelText.lastIndexOf(":")+1).trim());
+
+        if(_pointsInfo.length >= 10){
+
+          var _x1 = Number( _pointsInfo[1] ),
+            _y1 = Number( _pointsInfo[2] ),
+            _x2 = Number( _pointsInfo[8] ),
+            _y2 = Number( _pointsInfo[9] );
+
+          if(Math.abs(_x1 - _x2) > 0.01 || Math.abs(_y1 - _y2) > 0.01){
+            _labelDatum.id = _labelID;
+            _labelDatum.name = _labelName;
+            _labelDatum.color = _color;
+            _labelDatum.parentID = v.opts.chartId;
+            _labelDatum.samples = _labelValue;
+            currentSampleSize += _labelValue;
+
+            if(maxLabelValue < _labelValue) {
+              maxLabelValue = _labelValue;
+            }
+            _labels.push(_labelDatum);
+          }
+          _labelID++;
+        }else{
+          // StudyViewUtil.echoWarningMessg("Initial Label Error");
+        }
+      });
+
+      return _labels;
+    }
+
+    function updatePieLabels() {
+      updateCurrentLabels();
+      initReactData();
+      updateTables();
+    }
+
+    function updateTables() {
```

Comment:
This function has been changed, please see https://github.com/zhx828/cbioportal/commit/9bb24a309792b0e4ecfeb1204cf2fe177a17fc42